simpleble/src/backends/windows/AdapterBase.cpp

```diff
@@ -38,9 +38,20 @@ AdapterBase::AdapterBase(std::string device_id)
         [this](const auto& w, const Advertisement::BluetoothLEAdvertisementReceivedEventArgs args) {
             advertising_data_t data;
             data.mac_address = _mac_address_to_str(args.BluetoothAddress());
+            std::string addr_type;
+            int16_t addr_type_enum = args.BluetoothAddressType();
+            if (addr_type_enum == 0) {
+                addr_type = "public";
+            } else if (addr_type_enum == 1) {
+                addr_type = "random";
+            } else {
+                addr_type = "unspecified";
+            }
+            data.address_type = addr_type;
             data.identifier = winrt::to_string(args.Advertisement().LocalName());
             data.connectable = args.IsConnectable();
             data.rssi = args.RawSignalStrengthInDBm();
+            data.tx_power = args.TransmitPowerLevelInDBm();
```

Comment:
Before reading out the value, you need to check if the value is present: https://learn.microsoft.com/en-us/uwp/api/windows.devices.bluetooth.advertisement.bluetoothleadvertisementpublisher.includetransmitpowerlevel?view=winrt-22621#windows-devices-bluetooth-advertisement-bluetoothleadvertisementpublisher-includetransmitpowerlevel

If it's not present, using the same default value for all other cases makes sense.